client/dashboard/sites/staging-site-sync-modal/index.tsx

```diff
@@ -0,0 +1,275 @@
+		enabled: !! productionSiteId,
+	} );
+	const productionSiteSlug = productionSite?.slug;
+	const productionSiteTitle = productionSite?.name;
+
+	const { data: stagingSite } = useQuery( {
+		...siteByIdQuery( stagingSiteId ?? 0 ),
+		enabled: !! stagingSiteId,
+	} );
+	const stagingSiteSlug = stagingSite?.slug;
+	const stagingSiteTitle = stagingSite?.name;
+
+	const targetSiteSlug = targetEnvironment === 'production' ? productionSiteSlug : stagingSiteSlug;
+
+	const sourceSiteTitle = sourceEnvironment === 'staging' ? stagingSiteTitle : productionSiteTitle;
+	const targetSiteTitle =
+		targetEnvironment === 'production' ? productionSiteTitle : stagingSiteTitle;
+
+	// TODO: Uncomment once we need it.
+	// const querySiteId = sourceEnvironment === 'staging' ? stagingSiteId : productionSiteId;
+
+	const handleClose = useCallback( () => {
+		onClose();
+	}, [ onClose ] );
+
+	const handleConfirm = () => {
+		// Sync everything
+		// TODO: Uncomment once pullFromStaging and pushToStaging are available
+		// const include_paths = '';
+		// const exclude_paths = '';
+
+		if (
+			( syncType === 'pull' && environment === 'production' ) ||
+			( syncType === 'push' && environment === 'staging' )
+		) {
+			// TODO: Replace with the correct pullFromStaging (once it is moved to V2)
+			// pullFromStaging( { types: 'paths', include_paths, exclude_paths } );
+		} else {
+			// TODO: Replace with the correct pushToStaging (once it is moved to V2)
+			// pushToStaging( { types: 'paths', include_paths, exclude_paths } );
+		}
+	};
+
+	const handleDomainConfirmation = useCallback(
+		( value: string | undefined ) => setDomainConfirmation( value || '' ),
+		[]
+	);
+
+	const showDomainConfirmation = targetEnvironment === 'production';
+
+	const isSubmitDisabled = showDomainConfirmation && domainConfirmation !== productionSiteSlug;
+
+	return (
+		<Modal
+			title={ syncConfig[ environment ].title }
+			onRequestClose={ handleClose }
+			style={ { maxWidth: '668px' } }
+		>
+			<VStack spacing={ 5 }>
+				<Text>
+					{ createInterpolateElement( syncConfig[ environment ].description, {
+						a: <ExternalLink href={ `/backup/${ targetSiteSlug }` } children={ null } />,
+					} ) }
+				</Text>
+				<HStack spacing={ 4 } alignment="left">
+					<EnvironmentLabel
+						label={ syncConfig.fromLabel }
+						environmentType={ sourceEnvironment }
+						siteTitle={ sourceSiteTitle }
+					/>
+					<DirectionArrow />
+					<EnvironmentLabel
+						label={ syncConfig.toLabel }
+						environmentType={ targetEnvironment }
+						siteTitle={ targetSiteTitle }
+					/>
+				</HStack>
+				<VStack spacing={ 6 }>
+					{ showDomainConfirmation && (
+						<InputControl
+							__next40pxDefaultSize
+							label={
+								<HStack style={ { textTransform: 'none' } } alignment="left" spacing={ 1 }>
+									<Text>
+										{ __( 'Enter your site‘s name' ) }{ ' ' }
+										<Text color="var(--dashboard__foreground-color-error)">
```

Comment:
I have used slightly different color than in the original revamped modal. There we use `--studio-red-50`. If we decide, we could add this variable to V2 as well.

Reply:
I think this is OK, it's not a noticeable change.

| v1 | v2 |
|--------|--------|
| <img alt="Image" src="https://github.com/user-attachments/assets/0b599b1f-0b2c-4231-8731-3da7c4dacf8e" /> | <img alt="Image" src="https://github.com/user-attachments/assets/113e819c-6b07-4a89-8dc2-7aea20ae66a2" /> |